Document/0x05b-Basic-Security_Testing.md

```diff
@@ -362,60 +369,68 @@ Comprehensive documentation, including an installation guide, tutorials, and usa
-
-##### On-device Shell App
+One of the most common things you do when testing an app is accessing the device shell. In this section we'll see how to access the Android shell both and remotely from your host computer with/without a USB cable and from the device itself.
 
 ##### Remote Shell
 
-In order to connect to the shell of an Android device, [adb](https://developer.android.com/studio/command-line/adb "Android Debug Bridge") is usually your first tool of choice (unless you've configured remote SSH access).
-
-###### Connecting to an Android Physical Device
+In order to connect to the shell of an Android device from your host computer, [adb](https://developer.android.com/studio/command-line/adb "Android Debug Bridge") is usually your first tool of choice (unless you prefer to use remote SSH access, e.g. [via Termux](https://wiki.termux.com/wiki/Remote_Access#Using_the_SSH_server "Using the SSH server")).
 
-When connecting to an Android physical device, you must enable Developer Mode and USB debugging on the device in order to use the ADB debugging interface. Since Android 4.2, the "Developer options" sub menu in the Settings app is hidden by default. To activate it, tap the "Build number" section of the "About phone" view seven times. Note that the build number field's location varies slightly by device—for example, on LG Phones, it is under "About phone -> Software information." Once you have done this, "Developer options" will be shown at bottom of the Settings menu. Once developer options are activated, you can enable debugging with the "USB debugging" switch.
-
-Once USB debugging is enabled, connected devices can be viewed with the following command:
+For this section we assume that you've properly enabled Developer Mode and USB debugging as explained in "Testing on a Real Device". Once you've connected your Android device via USB, you can access the remote device's shell by running:
 
 ```shell
-$ adb devices
-List of devices attached
-BAZ5ORFARKOZYDFA    device
+$ adb shell
 ```
 
-Access to the remote device's shell
+> press Control + D or type `exit` to quit
+
+If your device is rooted or you're using the emulator, you can get root access by running `su` once in the remote shell:
 
 ```shell
 $ adb shell
+generic_x86:/ $ su
+generic_x86:/ # id
+uid=0(root) gid=0(root) groups=0(root),... context=u:r:su:s0
+
+$ adb shell
+bullhead:/ $ su
+bullhead:/ # id
+uid=0(root) gid=0(root) groups=0(root) context=u:r:su:s0
 ```
 
-> When you are ready to exit the remote shell, press Control + D or type `exit`.
+> Only if you're working with an emulator you may alternatively restart adb with root permissions with the command `adb root` so next time you enter `adb shell` you'll have root access already and it also allows to transfer data from/to file system locations where only the root user has access to (via `adb push/pull`). See more about data transfer in section "Host-Device Data Transfer" below.
```

Comment:
root access already and it also allows to transfer data from/to file system locations where only the root user has access to (via `adb push/pull`). See more about data transfer in section "Host-Device Data Transfer" below.

-> 

 root access already. This also allows to transfer data bidirectionally between your workstation and the iOS file system, even with access to locations where only the root user has access to (via `adb push/pull`). See more about data transfer in section "Host-Device Data Transfer" below.

Reply:
nice, done